flycheck-elsa.el

```diff
@@ -46,28 +46,96 @@
+The value can also be the symbol `cask' or `host'.
+
+If value is `cask', then Cask will be used to start Elsa on the
+current file.  This means that the file's project must have a Cask
+configuration file including Elsa as a development dependency.
+
+If value is `host', then Elsa and its dependencies must be
+installed and in Emacs' `load-path' as they will be used by
+flycheck.
+
+If value is a list of strings, it will be passed unchanged as the
+`:command' argument to `flycheck-define-command-checker'."
+  :group 'flycheck-elsa
+  :type '(choice
+          (const :tag "Cask" 'cask)
+          (const :tag "Emacs" 'host)
+          (repeat 'string)))
 
-We require that the project is managed by Cask and that Elsa is
-listed as a dependency."
-  (when (and (buffer-file-name)
-             (not (seq-find (lambda (f) (string-match-p f (buffer-file-name)))
-                            flycheck-elsa-ignored-files-regexps)))
-    (when-let ((cask-file (locate-dominating-file (buffer-file-name) "Cask")))
-      (let ((bundle (cask-initialize (file-name-directory cask-file))))
-        (cask-find-dependency bundle 'elsa)))))
+(defun flycheck-elsa--filename-ignored-p (filename)
+  "Return non-nil if FILENAME is matched by one of `flycheck-elsa-ignored-files-regexps'."
+  (seq-find
+   (lambda (regexp) (string-match-p regexp filename))
+   flycheck-elsa-ignored-files-regexps))
+
+(defun flycheck-elsa--enable-p ()
+  "Return non-nil if we can enable Elsa in current buffer."
+  (if (flycheck-elsa-cask-p)
+      (when (and (buffer-file-name)
+                 (not (flycheck-elsa--filename-ignored-p (buffer-file-name))))
+        (when-let ((cask-file (locate-dominating-file (buffer-file-name) "Cask")))
+          (let ((bundle (cask-initialize (file-name-directory cask-file))))
+            (cask-find-dependency bundle 'elsa))))
+    (if (buffer-file-name)
+        (not (flycheck-elsa--filename-ignored-p (buffer-file-name)))
+      t)))
 
 (defun flycheck-elsa--working-directory (&rest _)
   "Return the working directory where the checker should run."
   (if (buffer-file-name)
-      (when-let (file (locate-dominating-file (buffer-file-name) "Cask"))
-        (file-name-directory file))
+      (if (flycheck-elsa-cask-p)
+          (when-let (file (locate-dominating-file (buffer-file-name) "Cask"))
+            (file-name-directory file))
+        (file-name-directory (cdr (project-current))))
     default-directory))
 
+(defun flycheck-elsa--cask-command ()
+  "Return a list of strings to start Elsa within a Cask project."
+  '("cask" "exec" "elsa"))
+
+(defun flycheck-elsa--elsa-dependency-directories ()
+  "Return a list of directories where Elsa and its dependencies are installed."
+  (let ((directories (mapcar
+                      (lambda (lib)
+                        (file-name-directory (find-library-name lib)))
+                      '("elsa" "dash" "trinary" "f" "s" "flycheck"))))
```

Comment:
This has quite a potential for breaking if we forget to update the list here when we release new Elsa version.  Could it maybe read the package description of Elsa and get the dependencies from there?

Package.el has some function to read the headers.

Reply:
> Could it maybe read the package description of Elsa and get the dependencies from there? Package.el has some function to read the headers.

I updated my code to read dependencies from Elsa's Cask file instead.